emle/models.py

```diff
@@ -1210,3 +1223,394 @@ def forward(self, atomic_numbers, charges_mm, xyz_qm, xyz_mm):
+                raise TypeError("'device' must be of type 'torch.device'")
+        else:
+            device = _torch.get_default_device()
+
+        if dtype is not None:
+            if not isinstance(dtype, _torch.dtype):
+                raise TypeError("'dtype' must be of type 'torch.dtype'")
+        else:
+            self._dtype = _torch.get_default_dtype()
+
+        if atomic_numbers is not None:
+            if not isinstance(atomic_numbers, _torch.Tensor):
+                raise TypeError("'atomic_numbers' must be of type 'torch.Tensor'")
+            # Check that they are integers.
+            if atomic_numbers.dtype is not _torch.int64:
+                raise ValueError("'atomic_numbers' must be of dtype 'torch.int64'")
+            self._atomic_numbers = atomic_numbers.to(device)
+        else:
+            self._atomic_numbers = None
+
+        # Call the base class constructor.
+        super().__init__(
+            alpha_mode=alpha_mode,
+            device=device,
+            dtype=dtype,
+            create_aev_calculator=True,
+        )
+
+        # Load the MACE model.
+        if mace_model is not None:
+            if mace_model.startswith("mace-off23"):
+                size = mace_model.split("-")[-1]
+                assert (
+                    size in ["small", "medium", "large"]
+                ), f"Unsupported MACE model: '{mace_model}'. Available MACE-OFF23 models are 'mace-off23-small', 'mace-off23-medium', 'mace-off23-large'"
+                self._mace = _mace_off(model=size, device=device, return_raw_model=True)
+            else:
+                # Assuming that the model is a local model.
+                if _os.path.exists(mace_model):
+                    self._mace = _torch.load(mace_model, map_location="device")
+                else:
+                    raise FileNotFoundError(f"MACE model file not found: {mace_model}")
+        else:
+            # If no MACE model is provided, use the default MACE-OFF23(S) model.
+            self._mace = _mace_off(model="small", device=device, return_raw_model=True)
+
+        # Compile the model.
+        self._mace = _e3nn_jit.compile(self._mace).to(self._dtype)
+
+        # Create the z_table of the MACE model.
+        self._z_table = [int(z.item()) for z in self._mace.atomic_numbers]
+
+        if self._atomic_numbers is not None:
+            # Get the node attributes.
+            node_attrs = self._get_node_attrs(self._atomic_numbers)
+            self.register_buffer("ptr", _torch.tensor([0, node_attrs.shape[0]], dtype=_torch.long, requires_grad=False),)
+            self.register_buffer("node_attrs", node_attrs.to(self._dtype))
+            self.register_buffer("batch",_torch.zeros(node_attrs.shape[0], dtype=_torch.long, requires_grad=False),)
+
+            print("Node attributes:", node_attrs.shape)
+            print("Batch:", self.batch.shape)
+            print("Ptr:", self.ptr.shape)
+            print(node_attrs)
```

Comment:
Not sure if these are from debugging, but could they be removed. I'd rather the module was as silent as possible. I imagine that it might be created automatically for a user via convenience functions, e.g. `sire.qm.mace`, so random info to stdout might be confusing. (I appreciate that `torchani`, and likely `mace`, print out their own stuff, but I like to try to limit unnecessary output where possible.) If you would like to allow the user to get an overview of the objects state, then maybe add a `__str__` method that can print things in a nicely formatted way?

Reply:
Oh yes, these are from debugging and crept in accidentally. I'll fix this.